e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js

```diff
@@ -279,41 +279,48 @@ describe(
 
       queryBuilderHeader().within(() => {
         cy.findByLabelText("Back to Sleep dashboard").click();
-        cy.get("@dashboard.all").should("have.length", 1);
-        cy.get("@dashcardQuery.all").should("have.length", 2);
       });
 
       getDashboardCard().within(() => {
         cy.findByText("Sleep card").should("be.visible");
       });
+
+      // dashboard is taken from the cache, no re-fetch
+      cy.get("@dashboard.all").should("have.length", 1);
+      // the query is triggered second time as first one never loaded - no value in the cache
+      cy.get("@dashcardQuery.all").should("have.length", 2);
     });
 
     it("should preserve filter value when navigating between the dashboard and the question", () => {
+      // could be a regular dashboard with card and filters
       createDashboardWithSlowCard();
       cy.get("@dashboardId").then(visitDashboard);
+      cy.wait("@dashcardQuery");
 
-      cy.get("@dashcardQuery.all").should("have.length", 1);
+      // initial loading of the dashboard with card
+      cy.get("@dashcardQuery.all").should("have.length", 2);
```

Comment:
How can this be 2? We've just loaded the dashboard so at this point there was only one request.

Reply:
The reason you're seeing this locally when you isolate the common describe block using `.only` is that the previous test leaks data.

Since the previous test never had the chance to load dashboard card before finishing, it registered the response in the next one. This might be a Cypress bug, or maybe they even intentionally set it up like that... it doesn't matter.

We should never ever adjust tests (expected Metabase behavior) based on how a certain library (or our environment) behaves.

Reply:
> We should never ever adjust tests (expected Metabase behavior) based on how a certain library (or our environment) behaves.

I totally agree with this!

Reply:
should be good now